### ISSUE TITLE
Error when installing joplin-desktop via Home Manager

### ISSUE THREAD

**Issue by mjarosie** [NixOS/nixpkgs#302614]
### Describe the bug

I'm getting the `value is null while a string was expected` error when trying to install `joplin-desktop` via Home Manager.

darwin-rebuild switch --flake ~/.config/nix-darwin                                                                                                                                                               

### Steps To Reproduce
1. Add the following to Home Manager config:
```
    joplin-desktop = {
      enable = true;
    };
````
2.  Run `darwin-rebuild switch` (e.g.: `darwin-rebuild switch --flake ~/.config/nix-darwin`)
3. The output:

```
building the system configuration...
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'darwin-system-24.05.20240404.f480f9d+darwin4.36524ad'
         whose name attribute is located at /nix/store/dfwrffdwrrybch69gawdj9qd1rnp7mnk-source/pkgs/stdenv/generic/make-derivation.nix:331:7

       … while evaluating attribute 'activationScript' of derivation 'darwin-system-24.05.20240404.f480f9d+darwin4.36524ad'

         at /nix/store/87clbsxcf2wawlhi49k96ffb805sa3v0-source/modules/system/default.nix:95:7:

           94|
           95|       activationScript = cfg.activationScripts.script.text;
             |       ^
           96|       activationUserScript = cfg.activationScripts.userScript.text;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: value is null while a string was expected
```

### Expected behavior
Joplin Desktop should be installed successfully.

### Additional context
I'm using Home Manager as a nix-darwin module.

### Notify maintainers

@zorrobert

### Metadata

```console
~ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.4.0, macOS 14.4.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `""`
 - nixpkgs: `/nix/store/yj20wqsmn7x89z6avx9syjwiirga9jpv-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by zarelit** [comment]
home-manager issues can be filed in the home-manager project at https://github.com/nix-community/home-manager/issues

I think this bug is https://github.com/nix-community/home-manager/issues/5222

**Comment by zorrobert** [comment]
I'm very sorry for the issue, I hope that it will be resolved with #5226.

**Comment by Aleksanaa** [comment]
Note an issue inside Nixpkgs.
